### PR TITLE
feat: support typedoc v0.26

### PR DESCRIPTION
**Pull request by filiphsps**

### CODE REVIEW

**Comment by halostatue** [comment]
@filiphsps: I suspect that your second commit (removing the included build files) is an error. While I don't have access to the settings for this repo, I would bet that the GitHub pages configuration is set to `@main:/doc` (more or less), and you will have removed the updated documentation.

I would also *personally* suggest using `"typedoc": "^0.25.13 || ^0.26.3"` for a `1.0.4` release, as that would not force people to upgrade typedoc by upgrading typedoc-material-theme.

File: src/index.tsx
Comment (halostatue) [inline src/index.tsx:1]: The changes in this file are not related to the version support updates and should not be included.

**Comment by filiphsps** [comment]
> Previous comments aside, Typedoc 0.26 is _not compatible_ with Typedoc 0.25 or earlier without changes.
> 
> This is what is required to make the theme compatible with the updated API.
> 
> ```diff
> diff --git a/lib/index.js b/lib/index.js
> index 5a8374bc49325a896da009330feefed822057dd4..cfc9f99543bb80084b67cd93850efcb9cbb589eb 100644
> --- a/lib/index.js
> +++ b/lib/index.js
> @@ -36,9 +36,16 @@ export function load(app) {
>  
>            }
>          ` }))));
> -    app.listenTo(app.renderer, RendererEvent.END, () => {
> +
> +    const onRenderEnd = () => {
>          const from = resolve(__dirname, "../assets/style.css");
>          const to = resolve(app.options.getValue("out"), "assets/material-style.css");
>          cpSync(from, to);
> -    });
> +    }
> +
> +    if (typeof app.listenTo === 'function') {
> +        app.listenTo(app.renderer, RendererEvent.END, onRenderEnd);
> +    } else {
> +        app.renderer.hooks.on(RendererEvent.END, onRenderEnd);
> +    }
>  }
> diff --git a/package.json b/package.json
> index 9a6336171a99c7396b1d18e462c25e9191b9d251..27e996accbc18b47b8bc32d7507c86c416c1e890 100644
> --- a/package.json
> +++ b/package.json
> @@ -49,11 +49,12 @@
>    },
>    "devDependencies": {
>      "@types/node": "^20.14.6",
> -    "typedoc": "^0.25.13",
> +    "typedoc": "^0.25.13 || ^0.26.3",
>      "typescript": "^5.5.2"
>    },
>    "peerDependencies": {
> -    "typedoc": "^0.25.3"
> +    "typedoc": "^0.25.13 || ^0.26.3"
> +
>    },
>    "engines": {
>      "node": ">=18.0.0",
> diff --git a/src/index.tsx b/src/index.tsx
> index 4c47c4918b72fd30fc06469e6e3750712c0cc1a3..031fc1149d0e8e86aee13e26e1f21ac896a400b9 100644
> --- a/src/index.tsx
> +++ b/src/index.tsx
> @@ -59,12 +59,18 @@ export function load(app: Application) {
>      </script>
>    ));
>  
> -  app.listenTo(app.renderer, RendererEvent.END, () => {
> +  const onRenderEnd = () => {
>      const from = resolve(__dirname, "../assets/style.css");
>      const to = resolve(
>        app.options.getValue("out"),
>        "assets/material-style.css",
>      );
>      cpSync(from, to);
> -  });
> +  }
> +
> +  if (typeof app.listenTo === 'function') {
> +    app.listenTo(app.renderer, RendererEvent.END, onRenderEnd);
> +  } else {
> +    app.renderer.hooks.on(RendererEvent.END, onRenderEnd);
> +  }
>  }
> ```

Or more logically we should use a major version bump instead of supporting a now unsupported version of typedoc (half related tangent: would be nice if typedoc used semver correctly).

File: src/index.tsx
Comment (filiphsps) [reply src/index.tsx:1]: That’s an issue with the prettier/eslint configuration in the repo, please direct the criticism to the correct party :)

**Comment by halostatue** [comment]
> Or more logically we should use a major version bump instead of supporting a now unsupported version of typedoc (half related tangent: would be nice if typedoc used semver correctly).

Strictly speaking? It is. I would argue (weakly) that typedoc-material-theme should not have gone 1.x until typedoc itself did.

When at the `0.x` stage, the difference between `0.x` and `0.x + 1` should be considered the same as `x.y` and `x + 1.y` when using semver.

In the Ruby community, Rake went from `0.9` to `10.0` because the main developer had been asked about a 1.0 release, and realized that he'd been thinking of everything without the `0.` in front of it, so made it real with Rake 10.

I’ve gone with the more conservative change in my local file patch to be able to upgrade typedoc.

**Comment by dmnsgn** [comment]
Thanks both for you contributions, I have released v1.1.0 with parts of your changes.
